mirdata/datasets/queen.py

```diff
@@ -0,0 +1,244 @@
+
+        self.title = os.path.basename(self._track_paths["sections"][0]).split(".")[0]
+
+    @utils.cached_property
+    def chords(self):
+        """ChordData: chord annotation"""
+        return load_chords(self.chords_path)
+
+    @utils.cached_property
+    def key(self):
+        """KeyData: key annotation"""
+        return load_key(self.keys_path)
+
+    @utils.cached_property
+    def sections(self):
+        """SectionData: section annotation"""
+        return load_sections(self.sections_path)
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            section_data=[(self.sections, None)],
+            chord_data=[(self.chords, None)],
+            key_data=[(self.key, None)],
+            metadata={"artist": "The Queen", "title": self.title},
+        )
+
+
+def load_audio(audio_path):
+    """Load a Queen audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=None, mono=True)
+
+
+def load_chords(chords_path):
+    """Load Queen format chord data from a file
+
+    Args:
+        chords_path (str): path to chord annotation file
+
+    Returns:
+        (utils.ChordData): loaded chord data
+
+    """
+    if chords_path is None:
+        return None
+
+    if not os.path.exists(chords_path):
+        raise IOError("chords_path {} does not exist".format(chords_path))
+
+    start_times, end_times, chords = [], [], []
+    with open(chords_path, "r") as f:
+        dialect = csv.Sniffer().sniff(f.read(1024))
+        f.seek(0)
+        reader = csv.reader(f, dialect)
+        for line in reader:
+            start_times.append(float(line[0]))
+            end_times.append(float(line[1]))
+            chords.append(line[2])
+
+    chord_data = utils.ChordData(np.array([start_times, end_times]).astype(np.float).T, chords)
+
+    return chord_data
+
+
+def load_key(keys_path):
+    """Load Queen format key data from a file
+
+    Args:
+        keys_path (str): path to key annotation file
+
+    Returns:
+        (utils.KeyData): loaded key data
+
+    """
+    if keys_path is None:
+        return None
+
+    if not os.path.exists(keys_path):
+        raise IOError("keys_path {} does not exist".format(keys_path))
+
+    start_times, end_times, keys = [], [], []
+    with open(keys_path, "r") as fhandle:
+        reader = csv.reader(fhandle, delimiter="\t")
+        for line in reader:
+            if line[2] == "Key":
```

Comment:
Just to check - It looks like this will only be true for one line so there will only be one key interval, is that right?

Reply:
mmm no it is annotated each local key ("the modulations"). However, in the beatles dataset silence is not taken into account. (I think becouse mir eval is not also taking into account silence )
see an example:
```
0.000	0.456	Silence
0.456	83.139	Key	Bb
83.139	108.519	Key	Eb
108.519	142.649	Key	Bb
142.649	183.411	Key	Eb
183.411	206.713	Key	A
206.713	339.870	Key	Eb
339.870	353.076	Key	F
353.076	358.293	Silence
```

Reply:
Ah ok, i see! Can you add a small comment to the code?

Reply:
perfect!!

Reply:
done!